From 4354ef06062b9bf209668d499de016522dc7a33f Mon Sep 17 00:00:00 2001
From: Martin Wilck <mwilck@suse.com>
Date: Wed, 11 Mar 2020 23:43:43 +0100
Subject: [PATCH 125/130] todaysFactsWidget: replace Clutter.TableLayout with
 Clutter.GridLayout

Clutter.TableLayout has been deprecated in Clutter 1.18 and is not available
in GNOME shell 3.36 any more. It needs to be replaced with Clutter.GridLayout.
---
 extension/widgets/todaysFactsWidget.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/extension/widgets/todaysFactsWidget.js b/extension/widgets/todaysFactsWidget.js
index deafbdc..9f327f6 100644
--- a/extension/widgets/todaysFactsWidget.js
+++ b/extension/widgets/todaysFactsWidget.js
@@ -45,7 +45,7 @@ class TodaysFactsWidget extends St.ScrollView {
         this.factsBox.set_vertical(true);
         this.facts_widget = new St.Widget({
             style_class: 'hamster-activities',
-            layout_manager: new Clutter.TableLayout(),
+            layout_manager: new Clutter.GridLayout(),
             reactive: true
         });
         this.factsBox.add(this.facts_widget);
@@ -177,7 +177,7 @@ class TodaysFactsWidget extends St.ScrollView {
         for (let fact of facts) {
             let rowComponents = constructRow.bind(this)(fact, ongoingFact, this._controller, this._panelWidget.menu);
             for (let component of rowComponents) {
-                layout.pack(component, rowComponents.indexOf(component), rowCount);
+                layout.attach(component, rowComponents.indexOf(component), rowCount, 1, 1);
             }
             rowCount += 1;
         }
-- 
2.26.1

